Core/Services/OfferService.cs

```diff
@@ -64,15 +68,27 @@ public async Task CreateOfferAsync(OfferCreateDTO offerCreate, string userId)
             ExceptionMethods.UserNullCheck(await _userManager.FindByIdAsync(userId));
 
             var offer = _mapper.Map<Offer>(offerCreate);
+
             offer.OfferCreatorId = userId;
             offer.CreationDate = DateTimeOffset.UtcNow;
             offer.IsClosed = false;
             offer.CreatorRoleId = await _offerRoleService.GetRoleByNameAsync(offerCreate.Role);
             offer.GoodCategoryId = await _goodCategoryService
                 .GetGoodCategoryByNameAsync(offerCreate.GoodCategory);
-            offer.OfferPointId = await _pointService.CreateAsync(offerCreate.Point);
 
-            await _offerRepository.AddAsync(offer);
+            var point = await _pointService.CreateAsync(offerCreate.Point);
+
+            ExceptionMethods.PointNullCheck(point);
+
+            offer.OfferPointId = point.Id;
+
+            var offerFromDb = await _offerRepository.AddAsync(offer);
+
+            ExceptionMethods.OfferNullCheck(offerFromDb);
+
+            point.OfferId = offerFromDb.Id;
```

Comment:
Can we write:
"point.OfferId = offer.Id"
and reject "offerFromDb", "ExceptionMethods.OfferNullCheck(offerFromDb)"?

Reply:
We cannot just write "point.OfferId = offer.Id", because we DO NOT know it`s id before it is added to the database.